enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,410 @@
+
+The existing TechPreview client-based functionality will be put behind a dedicated flag and eventually deprecated. While the client-based functionality is present, the plain `oc adm upgrade status` command may hint about using it when it detects that `UpdateStatus` API is not provided by the cluster.
+
+#### Update UX in Web Console
+
+The Web Console presently contains specialized code for update progress presentation, which we hope to replace by simply reading `UpdateStatus` API. Web Console has resource discovery, so when `UpdateStatus` is present, the update progress presentation will read `UpdateStatus` API content instead of the current specialized logic. When `UpdateStatus` is not present, Web Console will gracefully degrade and display minimal information obtained from `ClusterVersion` resource's `Progressing` condition and warn about the disabled `UpdateStatus` capability, preventing more details from being provided. Clusters with enabled Web Console but disabled `UpdateStatus` will have a slightly worse experience than now (where Web Console itself interprets `Node` and `MachineConfigPool` states to show progress bars for node update), but we expect that combination to be rare.
+
+We may consider building a Web Console plugin for cluster update-related matters if the information exposed by `UpdateStatus` API grows sufficiently rich and complex.
+
+### Risks and Mitigations
+
+The proposal to deliver the API and a controller that both manages the API and monitors the cluster (producing insights) before achieving consensus on the eventual modular system architecture risks that the existing API will not accommodate the future architecture well. We are making a tradeoff to deliver the API early to start providing value, which also allows us to learn about how and if such an API is really consumed. Early delivery directly addresses the risk of investing effort into building a much larger system that may not address real user needs.
+
+#### API Size and Footprint
+
+The instances of `UpdateStatus` API, as proposed in https://github.com/openshift/api/pull/2012, are expected to be quite large. A single `UpdateStatus` resource status is a tree of the following shape, with update insight structures as leaves:
+
+```
+status:
+  # two nodes on this level
+  controPlane:
+    ...
+    informers:
+    # one node per informer reporting control plane insights, typically ~2-5
+    - name: informer-1
+      insights:
+      # one leaf per insight
+  workerPools:
+  # one node per pool
+  - name: mcp-1
+    informers:
+    # one node per informer reporting pool insights, typically ~2-5
+    - name: informer-2
+      insights
+      # one leaf per insight
```

Comment:
Can you calculate the binary size of one of these leaves for a Node object, what would a complete Node insight look like? Under WorkerPool I'd have many entries, 1 per node right?